### PR TITLE
Run more of the tests that have been ignored or missed

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
This PR ensures more of the tests that have been ignored or missed are actually run:
- `Tests.Reproduce` was not running due to a missing dependency on `xunit.runner.visualstudio`
- There was a reproduce test lost under a `Reproduce` folder outside the test project
- Integration tests were only being run in the `Tests` project, not those in `Tests.Reproduce`
- Fix broken reproduce tests that hadn't been being run
- Run the tests related to "abstractions"
- Remove references to OpenDistro/ODFE as it has been completely archived and the downloads no longer function

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by MaxKsyunz** [comment]
@Xtansia do you know why `Tests.Reproduce.GitHubIssue4537.CanDeserializeSnapshotShardFailure` test failed in the Create Release Candidate workflow?

**Comment by Yury-Fridlyand** [comment]
There is a flaky test in `Reproduce`. I rerun the workflow and now it pass.

**Comment by Yury-Fridlyand** [comment]
Actually, there are more than 100 not executed integration tests and few dozens of deactivated UT and IT.
Just a fact, it is not a proposal to fix them in scope of one PR.

**Comment by Xtansia** [comment]
> Actually, there are more than 100 not executed integration tests and few dozens of deactivated UT and IT. Just a fact, it is not a proposal to fix them in scope of one PR.

Is this just related to tests with `[Skip]` attributes, or some larger group?

**Comment by Yury-Fridlyand** [comment]
Unfortunately, there is larger group.
Tests with `[Skip]` are deactivated due to some bugs or limitations of specific product version.

**Review by dblock — APPROVED** [review]

File: abstractions/src/OpenSearch.Stack.ArtifactsApi/ServerType.cs [hunk 1]
```diff
@@ -31,8 +31,6 @@ namespace OpenSearch.Stack.ArtifactsApi
 	public enum ServerType
 	{
 		OpenSearch,
-		OpenDistro,
-		ElasticSearch,
 		DEFAULT = OpenSearch
 	}
 }
```
Comment (Yury-Fridlyand) [inline abstractions/src/OpenSearch.Stack.ArtifactsApi/ServerType.cs:36]: Perhaps, you can remove this entire enum and all references. It was added just to support testing on OpenDistro.

File: abstractions/src/OpenSearch.OpenSearch.Managed/ConsoleWriters/LineOutParser.cs [hunk 2]
```diff
@@ -39,15 +39,10 @@ public class LineOutParser
...
 		public static LineOutParser From(ServerType serverType)
 		{
 			switch (serverType)
 			{
-				case ServerType.OpenDistro:
-				case ServerType.ElasticSearch:
-					return LineOutParser.OpenDistro;
 				case ServerType.OpenSearch:
 					return LineOutParser.OpenSearch;
 				default:
```
Comment (Yury-Fridlyand) [inline abstractions/src/OpenSearch.OpenSearch.Managed/ConsoleWriters/LineOutParser.cs:48]: You can simplify this as well. This switch-case block was added for OpenDistro testing only.

File: abstractions/src/OpenSearch.Stack.ArtifactsApi/ServerType.cs (hunk 1 above)
Comment (Xtansia) [reply abstractions/src/OpenSearch.Stack.ArtifactsApi/ServerType.cs]: I've completely removed the enum now

File: abstractions/src/OpenSearch.Stack.ArtifactsApi/Artifact.cs [hunk 3]
```diff
@@ -41,13 +41,7 @@ public class Artifact
...
 			string buildHash)
 		{
-			ServerType = product.ServerType;
-			switch (ServerType)
-			{
-				case ServerType.OpenSearch: ProductName = "opensearch"; break; // actually = product.ProductName
-				case ServerType.OpenDistro: ProductName = "opendistroforelasticsearch"; break;
-				case ServerType.ElasticSearch: ProductName = "elasticsearch"; break;
-			}
+			ProductName = "opensearch";
```
Comment (Yury-Fridlyand) [inline abstractions/src/OpenSearch.Stack.ArtifactsApi/Artifact.cs:44]: Maybe
```suggestion
			ProductName = product?.ProductName ?? "opensearch";
```

File: abstractions/src/OpenSearch.Stack.ArtifactsApi/OpenSearchVersion.cs [hunk 4]
```diff
@@ -87,43 +88,9 @@ public Artifact Artifact(Product product)
...
-			gitHash = null;
-			var tokens = passedVersion.Split(':');
-			if (tokens.Length < 2)
-				return false;
-			version = tokens[1].Trim();
-			gitHash = tokens[0].Trim();
-			return true;
-		}
+		public static OpenSearchVersion From(string managedVersionString) =>
+			// TODO resolve `latest` and `latest-x` for OpenSearchÏ
```
Comment (Yury-Fridlyand) [inline abstractions/src/OpenSearch.Stack.ArtifactsApi/OpenSearchVersion.cs:92]:
```suggestion
			// TODO resolve `latest` and `latest-x` for OpenSearch
```
Somebody could get in trouble with unicode

File: tests/Tests/Cluster/RootNodeInfo/RootNodeInfoApiTests.cs [hunk 5]
```diff
@@ -62,22 +62,13 @@ protected override void ExpectResponse(RootNodeInfoResponse response)
 			response.Version.Should().NotBeNull();
 			response.Version.Number.Should().NotBeNullOrWhiteSpace();
 			response.Version.BuildDate.Should().BeAfter(default);
-			if (TestConfiguration.Instance.ServerType == OpenSearch.Stack.ArtifactsApi.ServerType.OpenSearch)
-				response.Version.Distribution.Should().NotBeNullOrWhiteSpace();
-			else
-				response.Version.BuildFlavor.Should().NotBeNullOrWhiteSpace();
```
Comment (Yury-Fridlyand) [inline tests/Tests/Cluster/RootNodeInfo/RootNodeInfoApiTests.cs:68]: I think you can remove `BuildFlavor` from `NodeInfo` and `OpenSearchVersionInfo` too.

**Review by Yury-Fridlyand — APPROVED** [review]

File: abstractions/src/OpenSearch.Stack.ArtifactsApi/Artifact.cs (hunk 3 above)
Comment (Xtansia) [reply abstractions/src/OpenSearch.Stack.ArtifactsApi/Artifact.cs]: Have followed suggestions

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Pull request merged**